src/fidesops/api/v1/endpoints/policy_endpoints.py

```diff
@@ -155,7 +156,7 @@ def create_or_update_rules(
         verify_oauth_client,
         scopes=[scopes.RULE_CREATE_OR_UPDATE],
     ),
-    policy_key: str,
+    policy_key: FidesKey,
```

Comment:
Hm, a side-effect of defining this type as a `FidesKey` means that if our URL has a non-fides key element in it, they also would get an unhandled 500 error, not sure how to handle.

```
PUT /policy/my-bad-policy/rule
```

So this is making me think about how all this discussion we've had over the last day is because `FidesKey` validation throws a custom `FidesValidationError` which pydantic/fastapi have no built-in way to work with.  I'm wondering if we should override `FidesKey` within FidesOps to throw ValueErrors and avoid all these workarounds we've had to add, and fix this issue while we're at it.

Reply:
I'm OK with this approach. 4xx makes a lot more sense than a 5xx in this context, and it's pretty low overhead to just override `FidesKey` to throw `ValueError`